src/client.rs

```diff
@@ -821,6 +821,14 @@ where
                 message_result = read_message(&mut self.read) => message_result?
             };
 
+            if message[0] as char == 'X' {
```

Comment:
A previous PR had moved the admin handling to before the message code match, an admin client disconnect produces an error log so this moves the 'X' message code check to before the admin client handling